### ISSUE TITLE
Consider building an ES6 Module as well

### ISSUE THREAD

**Issue by mattparrilla** [cox-auto-kc/react-entypo#8]
Matt from Burlington, VT office here. Howdy team!

I've got a feature request: publish your package as an es6 module. I believe you can do this side by side with an es5 build ([nwb does](https://github.com/insin/nwb/blob/master/docs/guides/ReactComponents.md#ecmascript-6-es6-modules-build)).

The reason: tree shaking. Webpack 2 offers [tree shaking](https://webpack.js.org/guides/tree-shaking/), but in order to take advantage of the feature, the module needs to be published as an ES6 module.

Obviously this is a huge package as it includes a ton of SVGs. It would be great if our users only had to pay the cost of the icons they were seeing.

I think all it might take is updating your `.babelrc`: https://medium.com/modus-create-front-end-development/webpack-2-tree-shaking-configuration-9f1de90f3233#b26a

**Disclaimer:** this is my first foray into webpack 2 and I don't know much about publishing modules. I just created a component library and was surprised that my library clocked in at >1MB and was further surprised to see that 450k of that comes from `react-entypo`.

BTW, I found this library organically, meaning Google, while looking for a way to get the new tracksuit icons in React. Was surprised to see you guys as the owners. Nice work!

**Comment by jasonhale** [comment]
Hi Matt!
I totally get the issue you're having.  May I ask how you're importing the icons you need?
If you import the icons _specifically_ they shouldn't take up so much space.
Like so:
```
import EntypoArrowBoldRight from 'react-entypo/lib/entypo/ArrowBoldRight';
import EntypoCalculator from 'react-entypo/lib/entypo/Calculator';
```

We recommend this approach.  I see the examples aren't very good at explaining that and we can work on updating them to be more specific.

Does that help?

**Comment by mattparrilla** [comment]
TL;DR: feel free to kill this issue! ES6 modules aren't all the way there yet and above solution works well enough

Though above solution works, I was still interested in experimenting with webpack tree shaking and so looked into getting a PR together that also built an ES6 module (basically transpile all code down into ES5, but use ES6 import/export instead of require). However upon investigating it looks like support for this sort of thing isn't official yet:
https://github.com/nodejs/node-eps/blob/4217dca299d89c8c18ac44c878b5fe9581974ef3/002-es6-modules.md#51-determining-if-source-is-an-es-module

and there are some recent reported issues with the not-yet-official approach:
https://github.com/infernojs/inferno/issues/928

So, someday!

**Comment by htchaan** [comment]
It can simply be done with [Webpack2's built-in tree shaking feature](https://webpack.js.org/guides/tree-shaking/) for es2015 modules. All you need to do is create a custom `lib/react-entypo.js` in your repo:

``` js
export Entypo500px from 'react-entypo/lib/entypo/500px';
export Entypo500pxWithCircle from 'react-entypo/lib/entypo/500pxWithCircle';
export EntypoAddToList from 'react-entypo/lib/entypo/AddToList';
// full list `ls node_modules/react-entypo/lib/entypo | pbcopy`
...
```

Then inside `webpack.config.js`, redirect your 'react-entypo' imports:

```js
externals: {
  'react-entypo': path.join(__dirname, 'lib/react-entypo'),...
```

That is it.

I tried a plugins for commonjs lib shaking, [webpack-common-shake](https://github.com/indutny/webpack-common-shake), but somehow it failed to shake. It's probably because the babel-transformed code at `node_modules/react-entypo/lib/index.js` is too complex for the plugin to handle.

Would be nice if we can just remove [`src/`](https://github.com/cox-auto-kc/react-entypo/blob/develop/.npmignore#L2) from [`.npmignore`](https://github.com/cox-auto-kc/react-entypo/blob/develop/.npmignore) providing users an option to import es code directly.
